File: Dev/C++/Mascaret/src/VEHA/Kernel/Parameter.cpp
#include "VEHA/Kernel/Parameter.h"
#include "VEHA/Kernel/Classifier.h"

namespace VEHA
{

Parameter::Parameter(const string& name,const string& direction,shared_ptr<Classifier> type)
:Element(name),_direction(direction),
 _type(type)
{
}

Parameter::~Parameter()
{
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Kernel/Parameter.cs

using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class Parameter : Element
    {

        private Classifier type;
        public Classifier Type
        {
            get { return type; }
            set { type = value; }
        }


        private string direction = "";
        public string Direction
        {
            get { return direction; }
            set { direction = value; }
        }


        public Parameter(string name, string direction, Classifier type)
            : base(name)
        {
            this.type = type;
            this.direction = direction;
        }

    }
}